### PR TITLE
upgrade of build tools, gradle wrapper, sdk

### PR DESCRIPTION
**Pull request by realrudi**

### CODE REVIEW

**Comment by poisdeux** [comment]
You will need to update the [travis configuration](https://github.com/xbmc/Kore/blob/master/.travis.yml) as well.

BTW, any particular reason to upgrade?

**Comment by erudisch** [comment]
No, it's just to be up to date

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -141,4 +141,7 @@
     <dimen name="fab_normal_margin">8dp</dimen>
     <dimen name="fab_mini_margin">16dp</dimen>
 
+    <dimen name="remote_icon_size">84dp</dimen>
```
Comment (poisdeux) [inline app/src/main/res/values/dimens.xml]: These are not needed as Android will pick the ones closest to the specified smallest width. I even think these will never be used as we use smallest width specific dimensions.

File: build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.android.tools.build:gradle:3.2.1'
```
Comment (poisdeux) [inline build.gradle]: Upgrading breaks the unit tests as. For some reason the shared resources required by the unit tests are not found when using the newer gradle version. So unless there is a really good reason to upgrade, I'd postpone the upgrade and wait for a newer version that hopefully also supports shared resources.

File: app/build.gradle [hunk 3]
```diff
@@ -12,7 +12,7 @@ def getVersionName = { ->
 
 android {
     compileSdkVersion 26
```
Comment (poisdeux) [inline app/build.gradle]: Why not set this to 28 as well?

Comment (SyncedSynapse) [reply app/build.gradle]: What implications did you see? I didn't look very closely, but didn't noticed anything that might impact us...

Comment (poisdeux) [reply app/build.gradle]: Sorry, I usually update the `compileSdkVersion` together with the `targetSdkVersion`, so was actually referring to updating both. It compiles fine with 28 so updating the `compileSdkVersion` should be safe.

We might also want to set the `targetSdkVersion` to 28 as we just made a new release, there should be enough time to test if everything still works as intended.

**Comment by SyncedSynapse** [comment]
@poisdeux I think this should be closed, but we should try to update the Gradle version. Does the issue still occurs with newer versions?

**Comment by poisdeux** [comment]
Sorry, totally forgot about this one. Seems to be running fine now with 3.4 of the android gradle plugin.

See #640

**Pull request closed without merging**